dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/sms/SimpleEventSMSListener.java

```diff
@@ -98,95 +84,27 @@ public SimpleEventSMSListener(
-    }
-
-    CategoryOptionCombo aoc = categoryService.getCategoryOptionCombo(aocid.getUid());
-
-    if (aoc == null) {
-      throw new SMSProcessingException(SmsResponse.INVALID_AOC.set(aocid));
-    }
-
-    if (!programService.hasOrgUnit(program, orgUnit)) {
-      throw new SMSProcessingException(SmsResponse.OU_NOTIN_PROGRAM.set(ouid, progid));
-    }
+    TrackerImportParams params =
+        TrackerImportParams.builder()
+            .importStrategy(TrackerImportStrategy.CREATE_AND_UPDATE)
+            .build();
+    TrackerObjects trackerObjects = map(subm, user);
+    ImportReport importReport = trackerImportService.importTracker(params, trackerObjects);
 
-    List<Enrollment> enrollments;
-    try {
-      enrollments =
-          new ArrayList<>(enrollmentService.getEnrollments(null, program, EnrollmentStatus.ACTIVE));
-    } catch (ForbiddenException | BadRequestException | NotFoundException e) {
-      // TODO(tracker) Find a better error message for these exceptions
-      throw new SMSProcessingException(SmsResponse.UNKNOWN_ERROR);
+    if (Status.OK == importReport.getStatus()) {
+      return SmsResponse.SUCCESS;
     }
-
-    // For Simple Events, the Program should have one Enrollment
-    // If it doesn't exist, this is the first event, we can create it here
-    if (enrollments.isEmpty()) {
-      Enrollment enrollment = new Enrollment();
-      enrollment.setEnrollmentDate(new Date());
-      enrollment.setOccurredDate(new Date());
-      enrollment.setProgram(program);
-      enrollment.setStatus(EnrollmentStatus.ACTIVE);
-
-      manager.save(enrollment);
-
-      enrollments.add(enrollment);
-    } else if (enrollments.size() > 1) {
-      // TODO: Are we sure this is a problem we can't recover from?
-      throw new SMSProcessingException(SmsResponse.MULTI_PROGRAMS.set(progid));
-    }
-
-    Enrollment enrollment = enrollments.get(0);
-    Set<ProgramStage> programStages = enrollment.getProgram().getProgramStages();
-    if (programStages.size() > 1) {
-      throw new SMSProcessingException(SmsResponse.MULTI_STAGES.set(progid));
-    }
-    ProgramStage programStage = programStages.iterator().next();
-
-    List<Object> errorUIDs =
-        saveEvent(
-            subm.getEvent().getUid(),
-            orgUnit,
-            programStage,
-            enrollment,
-            aoc,
-            user,
-            subm.getValues(),
-            subm.getEventStatus(),
-            subm.getEventDate(),
-            subm.getDueDate(),
-            subm.getCoordinates());
-    if (!errorUIDs.isEmpty()) {
-      return SmsResponse.WARN_DVERR.setList(errorUIDs);
-    } else if (subm.getValues() == null || subm.getValues().isEmpty()) {
-      // TODO: Should we save the event if there are no data values?
-      return SmsResponse.WARN_DVEMPTY;
-    }
-
-    return SmsResponse.SUCCESS;
+    // TODO(DHIS2-18003) we need to map tracker import report errors/warnings to an sms
```

Comment:
One other point in favor of separating program events and tracker events API is that in this listener if the user send a tracker program, the validation error from the tracker importer will be that the enrollment is missing, instead of wrong program type.